diff --git a/MiniCPM-V-demo-Android/app/src/main/cpp/llama_jni.cpp b/MiniCPM-V-demo-Android/app/src/main/cpp/llama_jni.cpp
index 174de78..1d0a226 100644
--- a/MiniCPM-V-demo-Android/app/src/main/cpp/llama_jni.cpp
+++ b/MiniCPM-V-demo-Android/app/src/main/cpp/llama_jni.cpp
@@ -98,11 +98,16 @@ Java_com_example_minicpm_1v_1demo_LlamaEngine_init(JNIEnv *env, jobject /*unused

 extern "C"
 JNIEXPORT jint JNICALL
-Java_com_example_minicpm_1v_1demo_LlamaEngine_load(JNIEnv *env, jobject, jstring jmodel_path) {
+Java_com_example_minicpm_1v_1demo_LlamaEngine_load(JNIEnv *env,
+                                                   jobject,
+                                                   jstring jmodel_path,
+                                                   jint jgpu_layers) {
     llama_model_params model_params = llama_model_default_params();
+    model_params.n_gpu_layers = (int) jgpu_layers;

     const auto *model_path = env->GetStringUTFChars(jmodel_path, 0);
-    LOGi("%s: Loading model from: \n%s\n", __func__, model_path);
+    LOGi("%s: Loading model from: \n%s\nn_gpu_layers=%d",
+         __func__, model_path, model_params.n_gpu_layers);

     FILE *f = fopen(model_path, "rb");
     if (!f) {
